sourcegraph/sourcegraph
# SCIM: Add support for soft delete

When modifying users using PUT/PATCH via skim toggling the `active` property should have the effect of soft deleting and restoring the user.

This code first updates the SCIM logic to remove any user specific entity logic from the generic logic that would apply to any scim entity such as how to manipulate attributes.  The effect of this is that each possible type of change is separated, and runs a check on the attributes before/after the changes to determine if it has occurred and an action needs to be taken.  After this change I replaced the existing hard delete code for `active` changes with a soft delete/reactivate. 

## Test plan
![image](https://user-images.githubusercontent.com/6098507/231184772-2769c3df-1bfd-4b1f-8ece-df8ed7a92880.png)


### Manual Test
SCIM  POST (create user)
Set password
Sign in 

SCIM GET 
Verify user is listed and active

SCIM PATCH - active -> false
Verify user can not login 

SCIM GET 
Verify user listed with active = false

SCIM PATCH - active -> true
Verify User can login again (only using username)
